plugins/modules/dcnm_service_policy.py

```diff
@@ -0,0 +1,1962 @@
+                return True, True
+            else:
+                return False, True
+        else:
+            return False, False
+
+        return (retry, deployed)
+
+    def dcnm_sp_compare_policy_info(self, want, have):
+
+        """
+        Routine to compare service policies from self.want and self.have and return an appropriate
+        returen code.
+
+        Parameters:
+            want (dict): The SP payload information populated from playbook
+            have (dict): The SP payload information that already exists
+
+        Returns:
+            DCNM_SRP_MATCH (string): if 'want' and 'have' match exactly
+            DCNM_SRP_NO_MATCH (string): if given 'want and 'have differ
+            mismatch_reasons (list): a list identifying objects that differed
+        """
+
+        mismatch_reasons = []
+
+        if want["enabled"] != str(have["enabled"]).lower():
+            mismatch_reasons.append("DCNM_SP_ENABLED_MISMATCH")
+        if want["sourceVrfName"] != have["sourceVrfName"]:
+            mismatch_reasons.append("DCNM_SP_SRC_VRF_MISMATCH")
+        if want["destinationVrfName"] != have["destinationVrfName"]:
+            mismatch_reasons.append("DCNM_SP_DST_VRF_MISMATCH")
+        if want["sourceNetwork"] != have["sourceNetwork"]:
+            mismatch_reasons.append("DCNM_SP_SRC_NETWORK_MISMATCH")
+        if want["destinationNetwork"] != have["destinationNetwork"]:
+            mismatch_reasons.append("DCNM_SP_DST_NETWORK_MISMATCH")
+        if (
+            str(want["reverseEnabled"]).lower()
+            != str(have["reverseEnabled"]).lower()
+        ):
+            mismatch_reasons.append("DCNM_SP_REV_ENABLED_MISMATCH")
+        if want["policyTemplateName"] != have["policyTemplateName"]:
+            mismatch_reasons.append("DCNM_SP_POL_TEMPLATE_MISMATCH")
+        if want["nextHopIp"] != have["nextHopIp"]:
+            mismatch_reasons.append("DCNM_SP_NHOP_IP_MISMATCH")
+
+        # If reverseEnabled is False, the fields related to the reverse direction will not be
+        # present. Check that before comparing
+        rnhip = want.get("reverseNextHopIp")
+        if rnhip and rnhip != have.get("reverseNextHopIp"):
+            mismatch_reasons.append("DCNM_SP_REV_NHOP_IP_MISMATCH")
+
+        for key in want["nvPairs"].keys():
+            # key "REVERSE" will be 'true' or 'false' in 'have' where as it will be a boolean True or False
+            # in 'want'. Convert the value in 'have' to bool before comparing
+
+            if key == "REVERSE":
+                if (
+                    have["nvPairs"]["REVERSE"] == "True"
+                    or have["nvPairs"]["REVERSE"] == "true"
+                ):
+                    have["nvPairs"]["REVERSE"] = True
+                if (
+                    have["nvPairs"]["REVERSE"] == "False"
+                    or have["nvPairs"]["REVERSE"] == "false"
+                ):
+                    have["nvPairs"]["REVERSE"] = False
```

Comment:
Suggest the following simplification for the logic above:

```python
import re
if key == "REVERSE":
    if re.search(r'true', have["nvPairs"]["REVERSE"].lower())
        have["nvPairs"]["REVERSE"] = True
    if re.search(r'false', have["nvPairs"]["REVERSE"].lower())
        have["nvPairs"]["REVERSE"] = False
```

Reply:
or how about have["nvPairs"]["REVERSE].lower() == "true"

Reply:
Sure, that also works.